add: ash update (part 1)

NightDawnFox opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
![image](https://github.com/user-attachments/assets/be774da9-fcf3-4356-a8ae-d9eb4ac578a9)

Массивное обновление пеплоходцев, затрагивающее огромное количество вещей, добавляющее кучу новых предметов и несколько новых механик - полностью завершенное.
Полный список изменений (не включает в себя балансные правки нулевого дня) - https://docs.google.com/document/d/1MSmRG_aWKBn17UcfXzlWP2HQMPHuqDQZq7n72S4F2oI/edit?usp=sharing
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
https://discord.com/channels/617003227182792704/755125334097133628/1340372235130507335
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

## Демонстрация изменений
сорян ребята я заебался делать скрины и видео, тут ничего не будет, все в дискорде((
<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->

## Тесты
Несколько десятков запусков локалки и проверка в игре каждого изменения и добавления. 
Отловка багов небольшой командой старых рисоваров на их локалке.
<!-- Как вы тестировали свой код. Ревьюеру будет проще, если он будет знать какие действия вы совершали, проверяя свой ПР. -->

NightDawnFox committed:
1. Map

NightDawnFox committed:
2. Map Spawn

NightDawnFox committed:
3. Component

NightDawnFox committed:
4. Fishing rod

NightDawnFox committed:
5. Krill (yourself)

NightDawnFox committed:
6. Component

NightDawnFox committed:
7. Bait

NightDawnFox committed:
8. Fish

NightDawnFox committed:
8.1 First fixes

NightDawnFox committed:
8.2 second fixes

NightDawnFox committed:
8.3 Another fixes & sprites

NightDawnFox committed:
9.1 Ash crab

NightDawnFox committed:
10. dead horseman & shellfish

todo: броня с панцирника не готова, доделать.

NightDawnFox committed:
11. bottom eel

NightDawnFox committed:
12. Red devourer

NightDawnFox committed:
13. Magma hammerhead

NightDawnFox committed:
fuck

NightDawnFox committed:
14. All remaining fish

NightDawnFox committed:
14. bugfix

при тестах заметил

NightDawnFox committed:
15. bigger lake

В ходе тестов было выявлено то, что прошлая вариация гарантированного озера не имеет "глубоководных" тайлов. Так что река слегка расширяется

NightDawnFox committed:
16. bugfixes 2: electric boogaloo

NightDawnFox committed:
ммда

NightDawnFox committed:
боже

NightDawnFox committed:
17. better tribal category

NightDawnFox committed:
17.1 bugfix

NightDawnFox committed:
18. The last Item

Остались лишь крафты предметов

NightDawnFox committed:
unrelated: grublunch bugfix

NightDawnFox committed:
19. Sprites for scorched skull

NightDawnFox committed:
20. Sulfuric tramp inhands

NightDawnFox committed:
21. The last items

Осталась броня (она еще не нарисована)

NightDawnFox committed:
22. New ashwalkers den

NightDawnFox committed:
23. The last item (for real this time)

NightDawnFox committed:
24. New ash rune

NightDawnFox committed:
25. Massive update

Заготовка под печь (не доделана).
Небольшие багфиксы
Заготовка под эффекты пищи и длительное потребление.

NightDawnFox committed:
26. bugfixes

Теперь печь реально готовит, ахуй

NightDawnFox committed:
27. First food

NightDawnFox committed:
28. Legs, veggies & treat

NightDawnFox committed:
+1

NightDawnFox committed:
+2

NightDawnFox committed:
+3 +4

NightDawnFox committed:
+5 +6

NightDawnFox committed:
bf

NightDawnFox committed:
+7

NightDawnFox committed:
фиксы

NightDawnFox committed:
+ сразу много

осталось два

NightDawnFox committed:
предпоследнее

утром время было

NightDawnFox committed:
Последняя еда

NightDawnFox committed:
небольшие багфиксы

NightDawnFox committed:
Система красок на тело

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

Antoonij committed:
rituals refactor

Antoonij committed:
fix nr1

Antoonij committed:
fix nr2

Antoonij committed:
fix nr3

Antoonij committed:
пилим пилим пилим

Antoonij committed:
допил x2

Antoonij committed:
доработка ритуалов

Antoonij committed:
fix

Antoonij committed:
more fix

Antoonij committed:
one more.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

Antoonij committed:
merge conflict

Antoonij committed:
improve rituals code, grace of lazis as invokers

Antoonij committed:
fix

NightDawnFox committed:
КАК ГОТОВИТЬ МЯСО

NightDawnFox committed:
Уаааа

NightDawnFox committed:
лопата!

NightDawnFox committed:
temp

Antoonij committed:
signal for rituals

NightDawnFox committed:
Update ritual_object.dm

NightDawnFox committed:
Merge branch 'fishing-update' of https://github.com/NightDawnFox/Paradise into fishing-update

NightDawnFox committed:
ньюлайн

Antoonij committed:
change signal pos

NightDawnFox committed:
боже прекрасно.

NightDawnFox committed:
Перевод и новый вариабл описания (для эшель)

NightDawnFox committed:
Набор разных изменений

1. новые спрайты яиц (:flushed:)
2. Деревянная лопатка и деревянный культиватор
3. Перевод всех шахтерских инструментов
4. Деревянный ящик
5. Обновлены спрайты вагонетки
6. Полный перевод гемов
7. Реворк лампы - теперь гемы можно использовать для смены цвета. красиво..

NightDawnFox committed:
извлечь

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
Переводы и новая мебель

NightDawnFox committed:
факела + крепления для них

dageavtobusnick committed:
Ritual UI

NightDawnFox committed:
Merge pull request #15 from dageavtobusnick/fishing-update

Ritual UI

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
Merge branch 'fishing-update' of https://github.com/NightDawnFox/Paradise into fishing-update

NightDawnFox committed:
деревянный стеллаж

NightDawnFox committed:
быстрофиксы

NightDawnFox committed:
ПЕРЕВОД

NightDawnFox committed:
обновленное логово

NightDawnFox commented:
@PiotrTheTchaikowsky как дела

NightDawnFox committed:
финальные штрихи (часть 1)

Тотемы - целеуказатели для шамана
Перевод абилок.
Руна теперь генерирует случайные рунк при активации

NightDawnFox committed:
фиксы перевода (часть 1)

NightDawnFox committed:
рецепт печи из блоков песка

потом забуду

NightDawnFox committed:
Update fishing_rod.dm

NightDawnFox committed:
Финальные правки + перевод (часть 2)

1. Полная ревизия и фиксы еды
2. Полный перевод еды
3. Удаление нахуй полу-готового магмитового крашера
4. добавление большего количества ликвид гибсов лужам

NightDawnFox committed:
блядь

NightDawnFox committed:
баланс и фиксы

THEROADCONEMAN commented:
> @PiotrTheTchaikowsky как дела

Ой блять 
![Zorp](https://github.com/user-attachments/assets/d890a8fd-9821-4e8e-a1ac-e7976b21b8cc)

NightDawnFox commented:
> > @PiotrTheTchaikowsky как дела
> 
> Ой блять ![Zorp](https://private-user-images.githubusercontent.com/139582597/412565231-d890a8fd-9821-4e8e-a1ac-e7976b21b8cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkzODY3OTUsIm5iZiI6MTczOTM4NjQ5NSwicGF0aCI6Ii8xMzk1ODI1OTcvNDEyNTY1MjMxLWQ4OTBhOGZkLTk4MjEtNGU4ZS1hMWFjLWU3OTc2YjIxYjhjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxMlQxODU0NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzVjZjIzYWE0NjVmNjlkMDg0NDM2MmE5ODFkY2QxOTVkN2ZmYzY1YWRmZjBmMjNkZmQyOWVkNTQzM2VhN2YzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6Llx2tlKvAdD_wu5CM93WTmQ3WO8cHejzr1DJ7gfMUw)

ты сам попросил пингануть тебя в ветке чтобы следить.

NightDawnFox committed:
фиксы по жалобам

1. Лодки теперь медленнее + сзади нормально видны
2. благодать лазиса - только костяное копье
3. фикс рецепт гильотины
4. наживка более не заражается муравьями

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into fishing-update

NightDawnFox committed:
рекомпил бандла

NightDawnFox committed:
ПАТЧ СОВМЕСТИМОСТИ С СУРВАЙВАЛ АПДЕЙТОМ

1. Доделка логова эшей
2. Добавление хирургического саквояжа вместо 999 отдельных.
3. Фиксы лавалейка
4. Фиксы крафта
5. рецепты в угледревах
6. прочее-прочее

NightDawnFox committed:
без лодочек(

NightDawnFox committed:
НОРМАЛЬНЫЕ краски

NightDawnFox committed:
Финальные штрихи. Часть 1

1. Рецепты тотемов
2. Рецепты кроватей
3. Рецепты факела и держателей
4. рецепты деревянной хуйни
5. фиксы перевода

NightDawnFox committed:
раз два три четыре пять - с детства с запятыми я дружу

NightDawnFox committed:
Обновление звуков

пиздец.

NightDawnFox committed:
ИН ХЭНД СПРАЙТЫ

боже мой бляяя

NightDawnFox committed:
упс)

NightDawnFox committed:
ФИНАЛЬНЫЕ ИЗМЕНЕНИЯ

1. В логово добавлены семена хлопка
2. У ритуалов убрано большинство требуемых вещей
3. Шанс выпадения черепа с рыбы теперь 40% вместо гаранта
4. У рыбы есть шанс 20% сорваться
5. поуч шамана теперь держит в себе наживку + некоторые новые ингридиенты с рыбалки
6. охотничий пояс держит практически все предметы, добавляемые обновлением.
7. гильотина теперь бьет со звуком циркулярки
8. понижены шансы у червей
9. У головы всадника теперь есть показатели брони + можно есть в ней
10. броня из хрящей теперь делается с помощью кожи (в теории можно вообще не драться с фауной)
11. наживка с рыбы теперь тоже может и не выпасть
12. рыба "кусает" при ударе, повышен урон при ударе. У циркулярной рыбы смешной звук + 20 брута за удар
13. нормальные (маленькие) размеры у красок
14. Свиток готовки теперь ведет на вики (спасибо Ване)
15. циркулярное лезвие теперь можно использовать для операций
Вроде всё. Пиздец

SAADf603 reviewed (dismissed):
maps cheked

BeebBeebBoob commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -384,3 +384,25 @@
+	)
+
+/obj/effect/spawner/random_spawners/sixty_pc_skull
+	name = "60pc scorched_skull"
```
не 40pc? `/obj/item/clothing/head/scorched_skull = 40`

NightDawnFox replied:
Правка делалась впопыхах. Сейчас поправлю..

BeebBeebBoob commented on code/datums/components/simple_fishing.dm:
```diff
@@ -0,0 +1,30 @@
+	else
+		for(var/fish in subtypesof(/obj/item/lavaland_fish/shoreline))
+			var/obj/item/lavaland_fish/shoreline/shore_fish = fish
+			catchable_fish += shore_fish
```
Для каждого тайла лавы, каждый новый инит с калькуляцией какие рыбы можно ловить... Плохая оптимизация.

BeebBeebBoob replied:
Я думаю можно и без компонента обойтись. Ловить можно только в лаве, поэтому просто дай прок на deep_water дабы указывать, какие тип рыб ты можешь ловить
```dm
lava/proc/get_fish()
	if(deep_water)
		return pick(GLOB.deep_fish) // На той стороне кода, пусть new ретурнатый тип
	else
		return pick(GLOB.shore_fish)
```

BeebBeebBoob commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -666,6 +673,29 @@ GLOBAL_LIST_INIT(fake_brass_recipes, list(
+	throw_speed = 1
+	throw_range = 7
+	throwforce = 15
+	origin_tech = "materials=4;biotech=5"
```
Как-то жирно по тех?

NightDawnFox replied:
Да нет, это базовые технологии, закрываемые в первые 10 минут раунда. Такие же технологии дают внутренние органы (или кибернетические сердца, печатаемые на протолате из железа и стекла), не вижу особых проблем. 
Особенно учитывая, что для получения этих зубов нужно убить минимум три или четыре голиафа (для лодки).

BeebBeebBoob commented on code/game/objects/items/weapons/fishing_rod.dm:
```diff
@@ -0,0 +1,182 @@
+		DATIVE = "старой доброй удочке",
+		ACCUSATIVE = "старую добрую удочку",
+		INSTRUMENTAL = "старой доброй удочкой",
+		PREPOSITIONAL = "старой доброй удочке"
```
Старая добрая удочка но спрайт показывает его идеальное состояние. Может подправить и просто "Удчока"?

BeebBeebBoob commented on code/game/objects/items/weapons/fishing_rod.dm:
```diff
@@ -0,0 +1,182 @@
+		DATIVE = "удочке",
+		ACCUSATIVE = "удочку",
+		INSTRUMENTAL = "удочкой",
+		PREPOSITIONAL = "удочке"
```
А вот здесь наоборот, старая удочка

BeebBeebBoob commented on code/game/objects/items/weapons/fishing_rod.dm:
```diff
@@ -0,0 +1,182 @@
+	if(!proximity)
+		return
+
+	var/datum/component/simple_fishing/fish_component = target.GetComponent(/datum/component/simple_fishing)
```
Как я ранее писал про компонент, Тут можно проверку на лава турф, уточнение переменной deep_water а далее ловля рыбы

BeebBeebBoob commented on code/game/objects/items/weapons/whetstone.dm:
```diff
@@ -99,3 +99,21 @@
+	righthand_file = 'icons/mob/inhands/lavaland/fish_items_righthand.dmi'
+	item_state = "crab_shell"
+	increment = 2
+	uses = -1
```
Уверен что это не будет абузным своровать на станцию?

NightDawnFox replied:
я забыл поправить на конечное пользование (с соглашением со старыми авторами)

BeebBeebBoob commented on code/game/objects/items.dm:
```diff
@@ -1302,6 +1303,7 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/g
 	alpha = 0
 	transform = animation_matrix
 
+	SEND_SIGNAL(src, COMSIG_ATOM_TEMPORARY_ANIMATION_START, 3)
```
Надо уточнить `COMSIG_ITEM_DROP_PRE_ANIMATE`

NightDawnFox replied:
я не знаю что это...

BeebBeebBoob commented on code/modules/mining/ash_walkers/ash_rune.dm:
```diff
@@ -0,0 +1,108 @@
+	var/transforming = FALSE
+	///Is our rune activated? Another sanity check, love it
+	var/activated = FALSE
+
```
Можно сюда переместить всё связанное с `ash_rune_centre` из 70 строки

BeebBeebBoob commented on code/modules/mining/ash_walkers/torch_holder.dm:
```diff
@@ -0,0 +1,287 @@
+			if(H.gloves)
+				var/obj/item/clothing/gloves/G = H.gloves
+				if(G.max_heat_protection_temperature)
+					prot = (G.max_heat_protection_temperature > 360)
```
может fakel.get_heat()? У тёмных перчей 1500, а сам факел даёт 1000. Думаю разумнее так
+ это пример https://github.com/ss220-space/Paradise/blob/master220/code/game/machinery/doors/airlock_types.dm#L198
```suggestion
					prot = (G.max_heat_protection_temperature > fakel.get_heat())
```

BeebBeebBoob commented on code/modules/mining/ash_walkers/torch_holder.dm:
```diff
@@ -0,0 +1,287 @@
+		span_notice("вы установили [declent_ru(ACCUSATIVE)] на [on_wall.declent_ru(ACCUSATIVE)]."))
+	qdel(src)
+
+/obj/machinery/torch_holder
```
машинарии?

NightDawnFox replied:
Ради системы процессинга сделал + делал, опираясь на код ламп, ибо в целом они похожи по принципу работы

BeebBeebBoob replied:
Ну... Т.к. там нету ничего активного кроме пассивного сгорания топлива, можно заменить на таймер.
Если заранее потушить, таймер удаляется и записывается сколько там осталось смотря на время старта и конца. Сам таймер это истечение топлива.
Машинарии просто не подходит + ты используешь процесс `SSobj` хотя у всех машинарии свой `SSmachines`
Ну очень не подходит под это
Прошу, переделай в `/obj/structure`, а с процессом по возможности как с компонентами дабы не жрал цпушку

BeebBeebBoob commented on code/modules/mob/living/carbon/human/species/unathi.dm:
```diff
@@ -338,38 +338,42 @@ They're basically just lizards with all-around marginally better stats and fire
-		to_chat(user, span_warning("You don't have any free hands."))
+		to_chat(user, span_warning("Ваши руки заняты."))
 
 /datum/action/innate/anvil_finder
```
думаю и сам датум стоит переименовать

BeebBeebBoob commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -699,10 +699,13 @@
 		for(var/datum/reagent/consumable/C in reagents.reagent_list) //we add the nutrition value of what we're currently digesting
 			fullness += C.nutriment_factor * C.volume / (C.metabolization_rate * metabolism_efficiency * digestion_ratio)
 	if(user == src)
+		if(time_to_eat > 0 && !do_after(user, time_to_eat, max_interact_count = 1))
```
Наконец-то. Теперь нельзя просто засунуть целую пиццу в рот за секунду

BeebBeebBoob commented on code/modules/mob/living/simple_animal/hostile/mining/gutlunch.dm:
```diff
@@ -45,6 +45,11 @@
+/mob/living/simple_animal/hostile/asteroid/gutlunch/examine(mob/user)
+	. = ..()
+	if(udder)
+		. += span_notice("[src]'s [udder.name] is currently has [udder.reagents.total_volume] units of milk in it")
```
По взгляду можно понять сколько внутри унции молока. Глаза бога в помойку

BeebBeebBoob replied:
Если будет интерес, то можно примерно там на глаз дать

NightDawnFox replied:
Это больше QoL, потому что наугад с этим работать проблемно и не интересно. Их доить - и так непопулярное занятие.

BeebBeebBoob commented on code/modules/research/designs/mining_designs.dm:
```diff
@@ -116,8 +116,8 @@
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 10000, MAT_TITANIUM = 15000, MAT_SILVER = 2500, MAT_DIAMOND = 2500)
-	build_path = /obj/item/twohanded/fishingrod
+	materials = list(MAT_METAL = 1000, MAT_TITANIUM = 1500)
```
Очень дешёвый. 4X цена пожалуйста.

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
тык

PlayerUnknown14 commented:
![image](https://github.com/user-attachments/assets/363960b1-6f67-46ca-bb60-a1062052ee21)

BeebBeebBoob reviewed (changes requested):
Я ещё не закончил

NightDawnFox committed:
Фиксы по требованию

Компонент перепишу следующим коммитом

NightDawnFox committed:
Merge branch 'fishing-update' of https://github.com/NightDawnFox/Paradise into fishing-update

NightDawnFox committed:
Рыбалка теперь не через компонент

Бип попросил

NightDawnFox committed:
ой)

NightDawnFox committed:
always place -> unpickable

NightDawnFox committed:
Новая пачка фиксов/решений

1. Улучшение лавового озера
2. Изменение рецепта ножек (старый не работает увы)
3. правки в селезенках
4. Переписывание торч холдера под структуру и таймеры
5. Багфикс с лампами

NightDawnFox committed:
наверное так?

NightDawnFox committed:
делить на 10

NightDawnFox committed:
блядь

ой.

NightDawnFox committed:
баланс ишшуе
